systems/console-bff/report/datasource/report_api.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * Copyright (c) 2023-present, Ukama Inc.
+ */
+import { RESTDataSource } from "@apollo/datasource-rest";
+
+class ReportAPI extends RESTDataSource {
+  async getGeneratedPdfReport(
+    baseURL: string,
+    id: string
+  ): Promise<ArrayBuffer> {
+    try {
+      const response = await fetch(`${baseURL}/v1/pdf/${id}`, {
+        headers: { Accept: "application/pdf" },
+      });
+
+      if (!response.ok) {
+        throw new Error(`Failed to fetch PDF: ${response.status}`);
+      }
```

Comment:
you can use `this.get(url, {headers: { Accept: "application/pdf" }})` to follow the RESTDataSource convention